kubectl-plugin/pkg/util/generation/generation_test.go

```diff
@@ -0,0 +1,103 @@
+package generation
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestGenerateRayCluterApplyConfig(t *testing.T) {
+	testRayClusterYamlObject := RayClusterYamlObject{
+		ClusterName:    "test-ray-cluster",
+		Namespace:      "default",
+		RayVersion:     "2.9.0",
+		Image:          "rayproject/ray:2.9.0",
+		HeadCPU:        "1",
+		HeadMemory:     "5Gi",
+		WorkerGrpName:  "worker-group1",
+		WorkerReplicas: 3,
+		WorkerCPU:      "1",
+		WorkerMemory:   "5Gi",
+	}
+
+	result, err := testRayClusterYamlObject.GenerateRayClusterApplyConfig()
+	assert.Nil(t, err)
+
+	assert.Equal(t, testRayClusterYamlObject.ClusterName, *result.Name)
+	assert.Equal(t, testRayClusterYamlObject.Namespace, *result.Namespace)
+	assert.Equal(t, testRayClusterYamlObject.RayVersion, *result.Spec.RayVersion)
+	assert.Equal(t, testRayClusterYamlObject.Image, *result.Spec.HeadGroupSpec.Template.Spec.Containers[0].Image)
+	assert.Equal(t, resource.MustParse(testRayClusterYamlObject.HeadCPU), *result.Spec.HeadGroupSpec.Template.Spec.Containers[0].Resources.Requests.Cpu())
+	assert.Equal(t, resource.MustParse(testRayClusterYamlObject.HeadMemory), *result.Spec.HeadGroupSpec.Template.Spec.Containers[0].Resources.Requests.Memory())
+	assert.Equal(t, testRayClusterYamlObject.WorkerGrpName, *result.Spec.WorkerGroupSpecs[0].GroupName)
+	assert.Equal(t, testRayClusterYamlObject.WorkerReplicas, *result.Spec.WorkerGroupSpecs[0].Replicas)
+	assert.Equal(t, resource.MustParse(testRayClusterYamlObject.WorkerCPU), *result.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Resources.Requests.Cpu())
+	assert.Equal(t, resource.MustParse(testRayClusterYamlObject.WorkerMemory), *result.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Resources.Requests.Memory())
+}
+
+func TestConvertRayClusterApplyConfigToYaml(t *testing.T) {
+	testRayClusterYamlObject := RayClusterYamlObject{
+		ClusterName:    "test-ray-cluster",
+		Namespace:      "default",
+		RayVersion:     "2.9.0",
```

Comment:
nit: update test references to use 2.39.0

Reply:
done!